Resources/Computing/ParallelLibraries/ParallelLibrary.py

```diff
@@ -0,0 +1,60 @@
+""" Parallel Library
+
+Mainly used by Computing Elements to run several jobs/pilots in parallel within a same allocation.
+This is an abstract class and cannot be used directly.
+Use cases:
+
+- multiple nodes have been allocated and have to be used: a parallel library can execute the pilot wrapper on all the
+  nodes in parallel.
+- multiple cpus have been allocated and have to be used: a parallel library could dispatch several jobs on the cores
+  available (e.g. when sites have no external connectivity).
+
+Parallel libraries implementations have to inherit from this class and override generateWrapper() and processOutput().
+
+**Parameters**
+
+The Parallel Libraries have to be defined in a CE section via the configuration system. See the page about
+configuring :ref:`resourcesComputing` for where the options can be placed.
+
+WorkingDirectory:
+   Generally defined in the ComputingElement source code depending on how they deal with the executables.
+   Indicates where the parallel library wrapper script has to be defined and retrieved.
+
+"""
+from DIRAC import gLogger
+
+
+class ParallelLibrary(object):
+
+  def __init__(self, name, workingDirectory):
+    """ Initialize the parallel library
+
+    :param str workingDirectory: directory that should contain the executable and its inputs
+    """
+    self.workingDirectory = workingDirectory
```

Comment:
what if `workingDirectory == None`?

Reply:
This should not be `None`, I will correct that.

I wonder if the executable (the pilot wrapper) shouldn't be embedded in the `srun` wrapper: copy paste the content of the pilot wrapper inside the `srun` wrapper.
We would have only one file (the `srun` wrapper) instead of 2 (the `srun` wrapper + the pilot wrapper) and we could immediately delete the pilot wrapper after the submission (otherwise we would need to keep it when using a `LocalCE` and then we would need to make `find` calls to delete it...)

Any thought about this?

Reply:
no strong opinion.